File: legacy-runtime/src/main/java/com/xiaomi/clientreport/util/SPManager.java
package com.xiaomi.clientreport.util;

import android.content.Context;
import android.content.SharedPreferences;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/clientreport/util/SPManager.class */
public class SPManager {
    private static volatile SPManager sInstance;
    private Context mContext;

    private SPManager(Context context) {
        this.mContext = context;
    }

    public static SPManager getInstance(Context context) {
        if (sInstance == null) {
            synchronized (SPManager.class) {
                try {
                    if (sInstance == null) {
                        sInstance = new SPManager(context);
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        return sInstance;
    }

    public long getLongValue(String str, String str2, long j) {
        long j2;
        synchronized (this) {
            try {
                j2 = this.mContext.getSharedPreferences(str, 4).getLong(str2, j);
            } catch (Throwable th) {
                return j;
            }
        }
        return j2;
    }

    public String getStringValue(String str, String str2, String str3) {
        String string;
        synchronized (this) {
            try {
                string = this.mContext.getSharedPreferences(str, 4).getString(str2, str3);
            } catch (Throwable th) {
                return str3;
            }
        }
        return string;
    }

    public void setLongValue(String str, String str2, long j) {
        synchronized (this) {
            SharedPreferences.Editor editorEdit = this.mContext.getSharedPreferences(str, 4).edit();
            editorEdit.putLong(str2, j);
            editorEdit.commit();
        }
    }

    public void setStringnValue(String str, String str2, String str3) {
        synchronized (this) {
            SharedPreferences.Editor editorEdit = this.mContext.getSharedPreferences(str, 4).edit();
            editorEdit.putString(str2, str3);
            editorEdit.commit();
        }
    }
}

File: legacy/src/main/java/com/xiaomi/clientreport/util/SPManager.kt
package com.xiaomi.clientreport.util
import io.github.magisk317.mipush.protocol.model.*

import android.content.Context
import android.content.SharedPreferences

class SPManager private constructor(private val mContext: Context) {

    companion object {
        @Volatile
        private var sInstance: SPManager? = null

        fun getInstance(context: Context): SPManager {
            return sInstance ?: synchronized(SPManager::class.java) {
                sInstance ?: SPManager(context).also { sInstance = it }
            }
        }
    }

    fun getLongValue(str: String, str2: String, j: Long): Long {
        return try {
            mContext.getSharedPreferences(str, 4).getLong(str2, j)
        } catch (th: Throwable) {
            j
        }
    }

    fun getStringValue(str: String, str2: String, str3: String): String? {
        return try {
            mContext.getSharedPreferences(str, 4).getString(str2, str3)
        } catch (th: Throwable) {
            str3
        }
    }

    fun setLongValue(str: String, str2: String, j: Long) {
        synchronized(this) {
            val editorEdit: SharedPreferences.Editor =
                mContext.getSharedPreferences(str, 4).edit()
            editorEdit.putLong(str2, j)
            editorEdit.commit()
        }
    }

    fun setStringnValue(str: String, str2: String, str3: String) {
        synchronized(this) {
            val editorEdit: SharedPreferences.Editor =
                mContext.getSharedPreferences(str, 4).edit()
            editorEdit.putString(str2, str3)
            editorEdit.commit()
        }
    }
}
